pkg/csi/cephfsdaemonset.go

```diff
@@ -251,6 +254,9 @@ var cephFSDaemonSetSpec = appsv1.DaemonSetSpec{
 
 func SetCephFSDaemonSetDesiredState(ds *appsv1.DaemonSet) {
 	// Copy required labels
+	if ds.Labels == nil {
+		ds.Labels = make(map[string]string)
+	}
 	for key := range cephfsDaemonsetLabels {
 		ds.Labels[key] = cephfsDaemonsetLabels[key]
 	}
```

Comment:
This pattern where the label map does not exist when trying to set it keeps appearing. 
Let's refactor into a util which returns guarantee labels on an object.
```
func AddLabels(obj metav1.Object, newLabels) {
	labels := obj.GetLabels()
	if labels == nil {
	         labels = map[string]string{}
                 obj.SetLabels(labels) 
	}
	maps.Copy(labels, newLabels)
}
```

Then we can simply do
```suggestion
	utils.AddLables(ds.Labels, cephfsDaemonsetLabels)
```

Reply:
done.